cider/exec/template/IRCodegen.cpp

```diff
@@ -717,26 +717,58 @@ std::vector<JoinLoop> Executor::buildJoinLoops(
     bool has_remaining_left_join_quals =
         rem_left_join_quals_it != plan_state_->left_join_non_hashtable_quals_.end() &&
         !rem_left_join_quals_it->second.empty();
-    const auto outer_join_condition_remaining_quals_cb =
-        [this, level_idx, &co](const std::vector<llvm::Value*>& prev_iters) {
-          // when we have multiple quals for the left join in the current join level
-          // we first try to build a hashtable by using one of the possible qual,
-          // and deal with remaining quals as extra join conditions
-          FetchCacheAnchor anchor(cgen_state_.get());
-          addJoinLoopIterator(prev_iters, level_idx + 1);
-          llvm::Value* left_join_cond = cgen_state_->llBool(true);
-          CodeGenerator code_generator(this);
-          auto it = plan_state_->left_join_non_hashtable_quals_.find(level_idx);
-          if (it != plan_state_->left_join_non_hashtable_quals_.end()) {
-            for (auto expr : it->second) {
-              left_join_cond = cgen_state_->ir_builder_.CreateAnd(
-                  left_join_cond,
-                  code_generator.toBool(
-                      code_generator.codegen(expr.get(), true, co).front()));
+    std::function<llvm::Value*(const std::vector<llvm::Value*>&)>
+        outer_join_condition_remaining_quals_cb;
+    if (co.use_cider_data_format) {
+      outer_join_condition_remaining_quals_cb =
+          [this, level_idx, &co](const std::vector<llvm::Value*>& prev_iters) {
```

Comment:
why not implement this lambda as a normal function?
if the lambda convert to a `std::function<llvm::Value*(const std::vector<llvm::Value*>&)>`
the std::function's execution efficiency is really slow.

Reply:
this part would refactor in JITLib, so does not care about efficiency right now.